dataprofiler/tests/data_readers/test_data.py

```diff
@@ -160,6 +161,23 @@ def test_read_url_verify_ssl(self, mock_request_get):
         ):
             data_obj = Data("https://test.com")
 
+    @patch("boto3.client")
+    @patch(
+        "os.environ",
+        {
+            "AWS_ACCESS_KEY_ID": "<YOUR_ACCESS_KEY>",
+            "AWS_SECRET_ACCESS_KEY": "<YOUR_SECRET_KEY>",
+        },
+    )
+    def test_read_s3_uri(self, mock_boto3_client):
+        # Create a custom mock response
+        custom_response = {"Body": MagicMock()}
+        custom_response["Body"].read.return_value = b"Test S3 content"
+
+        # Mock the behavior of the S3 client to return the custom response
+        mock_boto3_client.return_value.get_object.return_value = custom_response
+        data_obj = Data("s3a://my-bucket/my_file.txt")
```

Comment:
- let's also do some validation on the `data_obj` with some `assert ...`
- or remove `data_obj` variable since its an un-used variable defined

Reply:
Just added an assert validation on `data_obj`